refactor: sanitize error object before emitting MESSAGE_RECEIVED_WITH…

ViaSocket-Git opened the pull request:
…_ERROR event

- Added explicit type annotation for error parameter in sendDataToAction catch block
- Created sanitizedError object with message, status, statusText, and data properties
- Added console.log for error debugging
- Pass sanitizedError instead of raw error to emitEventToParent to ensure consistent error structure

Commit:
refactor: add error handling and display for failed message sends

- Added error state to ChatState interface and initialChatState in chatReducerV2
- Added setError reducer action to update error state
- Exported setError action from chatSlice
- Updated useSendMessage to clear errors on new message and set error on API failure
- Modified sendDataToAction to return sanitized error message on failure
- Added error message display above input field in ChatbotTextField with warning icon
- Improved error handling in MESSAGE

Natwar589 commented on components/Interface-Chatbot/ChatbotTextField.tsx:
```diff
@@ -384,6 +385,16 @@ const ChatbotTextField: React.FC<ChatbotTextFieldProps> = ({ className, chatSess
+      {error && (
+        <div className="px-2 pb-2">
+          <div className="text-red-500 text-sm flex items-center gap-2">
+            <span className="text-red-500">⚠</span>
```
use icon from lucide react

Commit:
refactor: replace warning emoji with TriangleAlert icon in error message display

- Imported TriangleAlert icon from lucide-react
- Replaced warning emoji (⚠) with TriangleAlert component in error message display
- Applied consistent icon styling with w-4 h-4 text-error classes
